### DISCUSSION TITLE
How to user "smart query conditions" in many-to-many relations

### DISCUSSION THREAD

**Discussion by mmick66** [mikro-orm/mikro-orm#6250] [Q&A] [answered]
### Context

I have two entities:

```
@Entity({ tableName: 'students' })
export class StudentEntity {
  @Property()
  firstName: string;

  @Property()
  lastName: string;

  @ManyToMany({
  entity: () => TeacherEntity,
    pivotTable: 'students_to_teachers',
    joinColumn: 'student_id',
    inverseJoinColumn: ;teacher_id'
  })
  teachers = new Collection<TeacherEntity>(this);
}
```
and ...
```
@Entity({ tableName: 'teachers' })
export class TeacherEntity {
   @OneToOne(() => UserEntity, {
    owner: true,
    fieldName: 'user_id',
  })
  user: UserEntity;

  @ManyToMany({
    entity: () => StudentEntity,
    pivotTable: 'students_to_teachers',
    joinColumn: 'teacher_id',
    mappedBy: (s: StudentEntity) => s.teachers
  })
  students = new Collection<StudentEntity>(this);
}
```
### Task
Given a `userId` for a `UserEntity` of a `TeacherEntity`, I can take all the students who are connected to this teacher via
```
const students = await this.studentsRepo.find(
  {
    teachers: {
      user: {
        id: userId
      }
    }
  }
);
```
### Problem
Now I want the opposite. I want to take all students who **are not** connected with this teacher. Instinctively, I did something like
```
const students = await this.studentsRepo.find(
  {
    teachers: {
      user: {
        $ne: {
          id: userId
        }
      }
    }
  }
);
```
but it did not work. 

Please advise!

**Comment by B4nan** [comment, marked as answer]
answered on SO

https://stackoverflow.com/questions/79210118/how-to-user-smart-query-conditions-in-mikroorm-in-many-to-many-relations-6250/79210163

**Reply by mmick66** [reply]
Thank you so much for the quick response. You solution works!
* ps. how do you do this? not even 20 minutes had elapsed and your answer was there ;-)

**Reply by B4nan** [reply]
I've built this thing, so I know well how it works :] And as said in the SO post, I am subscribed to both of those channels, I am happy to answer immediately to anything I can easily understand and know/think how to resolve it.
